### PR TITLE
feat: geo functions

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

File: src/query/functions-v2/src/scalars/geo.rs [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        |lon1:F64,lat1:F64,lon2:F64,lat2:F64,_| {
+            F32::from(distance(lon1.0 as f32, lat1.0 as f32, lon2.0 as f32, lat2.0 as f32, GeoMethod::Wgs84Meters))
+        },
+    );
+
+    // great circle angle
+    registry.register_4_arg::<NumberType<F64>, NumberType<F64>, NumberType<F64>, NumberType<F64>,NumberType<F32>,_, _>(
+        "great_circle_angle",
+        FunctionProperty::default(),
+        |_,_,_,_|None,
```
Comment (sundy-li) [inline src/query/functions-v2/src/scalars/geo.rs:96]: can have min/max for the f32 domain.

**Review by sundy-li — APPROVED** [review]

**Comment by sundy-li** [comment]
need to fix the tests.

**Comment by BohuTANG** [comment]
Some tests are failed for a long time, please address :)

**Comment by ariesdevil** [comment]
@BohuTANG It seems the failed tests are not related to this PR.

**Comment by Xuanwo** [comment]
> @BohuTANG It seems the failed tests are not related to this PR.

Related to https://github.com/datafuselabs/databend/issues/7986

**Comment by xudong963** [comment]
> @BohuTANG It seems the failed tests are not related to this PR.

flaky test, related issue: https://github.com/datafuselabs/databend/issues/7986

I'm trying to fix it.

**Comment by BohuTANG** [comment]
@mergify update

**Pull request merged**